src/plugins/openInApp/index.ts

```diff
@@ -155,14 +145,12 @@ export default definePlugin({
         return false;
     },
 
-    handleAccountView(event: { preventDefault(): void; }, platformType: string, userId: string) {
-        if (platformType === "spotify" && settings.store.spotify) {
-            VencordNative.native.openExternal(`spotify:user:${userId}`);
-            event.preventDefault();
-        } else if (platformType === "steam" && settings.store.steam) {
-            VencordNative.native.openExternal(`steam://openurl/https://steamcommunity.com/profiles/${userId}`);
-            showToast("Opened link in Steam", Toasts.Type.SUCCESS);
-            event.preventDefault();
+    handleAccountView(_, platformType: string, userId: string) {
+        for (const [key, rule] of Object.entries(urlReplacementRules)) {
+            if (key === platformType && rule.accountViewReplace && pluginSettings.store[key]) {
```

Comment:
no need for a loop here :P 
```suggestion
        const rule = urlReplacementRules[platformType];
        if (rule?.accountViewReplace && pluginSettings.store[platformType]) {
```